src/test/java/io/aiven/kafka/connect/opensearch/OpensearchSinkTaskTest.java

```diff
@@ -51,4 +70,22 @@ void convertTopicToIndexName() {
 
     }
 
+    @Test
+    void failToStartWhenReportIsConfiguredAndErrantRecordReporterIsMissing(final @Mock SinkTaskContext context) {
+
+        final var props = Map.of(
+                CONNECTION_URL_CONFIG, "http://localhost",
+                MAX_BUFFERED_RECORDS_CONFIG, "100",
+                MAX_IN_FLIGHT_REQUESTS_CONFIG, "5",
+                BATCH_SIZE_CONFIG, "2",
+                LINGER_MS_CONFIG, "1000",
+                MAX_RETRIES_CONFIG, "3",
+                READ_TIMEOUT_MS_CONFIG, "1",
+                BEHAVIOR_ON_VERSION_CONFLICT_CONFIG, BulkProcessor.BehaviorOnMalformedDoc.REPORT.toString()
+        );
+        final var task = new OpensearchSinkTask();
+        task.initialize(context);
```

Comment:
Should we explicitly set up an expectation for `context::errantRecordReporter` to return `null`, to make the intent of this test clearer?

Also, can we add a case where `context::errantRecordReporter` throws a `NoSuchMethodError` and handles it gracefully (i.e., in the same way it does in this test case)?